index.js

```diff
@@ -1,18 +1,57 @@
 const { question } = require("readline-sync");
-const { displayWordSoFar, isGameWon, isGameLost } = require("./gamelogic");
-
+const { displayWordSoFar, isGameWon, isGameLost, showHangman } = require("./gamelogic");
+//  countFaults geeft aan hoeveel foute antwoorden zijn gegegeven
+let countFaults=0;
 function game(word, guesses) {
   console.log("Dit heb je tot nu toe geraden: ", guesses);
 
-  const letter = question("Raad een letter: ");
+  //  check of game al gewonnen is
+  isGameWon(word,guesses);
+
+  //  check of game al verloren is
+  isGameLost(word,guesses);
+
+  //  check of ingevoerde letter wel/niet voorkomt in het woorden geeft string met juist geradeletters en nog niet gerade letters(streepjes)
+  displayWordSoFar(word,guesses);
+
+  // deze variable bevat de waarde van displayWordSoFar voor de nieuwe input
+  let tempResult=displayWordSoFar(word,guesses);
+
+  //  input wordt automatisch omgezet naar lowerCase
+  const letter = question("Raad een letter: ").toLowerCase();
+
+  //  check of de invoer uit 1 karakter bestaat
+  if (letter.length == 1) {
+    //  check of ingevoerde letter niet voorkomt in de guesses-array
+    if (!guesses.includes(letter)) {
+      guesses.push(letter);
+      //  check of tempResult gelijk is aan displayWordSoFar, zo ja is er geen nieuwe letter goed geraden, dus faultCounter +1
+      if (tempResult == displayWordSoFar(word, guesses)) {
+        countFaults += 1;
+      }
+    } else {
+      console.log('Helaas, dit karakter heb je al eerder ingevoerd!');
+    }
 
-  // voeg de geraden letter toe aan de array met guesses
-  guesses.push(letter);
+    //  functie die de galg tekent in de console;
+    showHangman(countFaults);
+    console.log(`Foute antwoorden: ${countFaults}`);
+  } else {
+    console.log("Sorry, je mag maar 1 karakter tegelijk invoeren");
+  }
+
+
+
+
+
+  //  als game gewonnen of verloren is, stop de game
+  if((isGameWon(word,guesses))||(isGameLost(word,guesses))){
+    return;
+  }
 
```

Comment:
Alle gemaakte bonusopdrachten even samen gepakt. Zagen er allemaal prima uit, wederom niets op aan te merken. Werkt perfect en is goed kort geschreven met hele duidelijke comments.